src/features/admin/routes/item/item-variant-form.tsx

```diff
@@ -0,0 +1,116 @@
+import {
+  Box,
+  Checkbox,
+  FormControl,
+  FormControlLabel,
+  InputLabel,
+  MenuItem,
+  Select,
+  Stack,
+  Typography,
+} from "@mui/material";
+import {
+  Controller,
+  FieldArrayWithId,
+  useFieldArray,
+  useFormContext,
+} from "react-hook-form";
+import { ItemData } from "../../types/table-data";
+import { TransformController } from "../../../../components/transform-controller";
+import { ChangeEvent, useEffect } from "react";
+
+type ItemVariantFormProps = {
+  variant: FieldArrayWithId<ItemData, "availableVariants", "variantControlId">;
+  index: number;
+};
+
+function ItemVariantForm({ variant, index }: ItemVariantFormProps) {
+  const { control, watch, setValue } = useFormContext<ItemData>();
+
+  const variantName =
+    `availableVariants.${index}` as `availableVariants.${number}`;
+  const { fields } = useFieldArray({
+    control,
+    name: `${variantName}.variants`,
+    keyName: "singleVariantControlId",
+  });
+
+  const selectedSingleVariants = watch(`${variantName}.variants`).filter(
+    ({ selected }) => selected
+  );
+  const selectedSingleVariantsElements = selectedSingleVariants.map(
+    (singleVariant) => (
+      <MenuItem key={singleVariant.id} value={singleVariant.id}>
+        {singleVariant.name}
+      </MenuItem>
+    )
+  );
+
+  const isNoSingleVariantSelected = !selectedSingleVariants.length;
+  useEffect(() => {
+    if (isNoSingleVariantSelected) {
+      setValue(`${variantName}.defaultVariantId`, 0);
+    }
+  }, [isNoSingleVariantSelected]);
+
+  return (
+    <Stack spacing={2}>
+      <Box display="flex" alignItems="baseline" justifyContent="space-between">
+        <Typography variant="h5">{variant.name}</Typography>
+        <FormControl sx={{ minWidth: 200 }}>
+          <InputLabel id={`variant-${variant.variantControlId}-default-label`}>
+            Standard auswählen
+          </InputLabel>
+          <TransformController
+            name={`${variantName}.defaultVariantId`}
+            control={control}
+            transform={{
```

Comment:
Same for this transform.

Reply:
added todo - tried to move it to a constant but there were some type-errors and ...tbh i currently have not enough patience :D